document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -0,0 +1,205 @@
+package org.hypertrace.core.documentstore.postgres.utils;
+
+import static org.hypertrace.core.documentstore.Collection.UNSUPPORTED_QUERY_OPERATION;
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.CREATED_AT;
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.DOCUMENT;
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.DOC_PATH_SEPARATOR;
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.ID;
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.UPDATED_AT;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.hypertrace.core.documentstore.postgres.Params;
+import org.hypertrace.core.documentstore.postgres.Params.Builder;
+
+public class PostgresUtils {
+  private static final String QUESTION_MARK = "?";
+  private static final String JSON_FIELD_ACCESSOR = "->";
+  private static final String JSON_DATA_ACCESSOR = "->>";
+
+  private static final Set<String> OUTER_COLUMNS =
+      new HashSet<>() {
+        {
+          add(CREATED_AT);
+          add(ID);
+          add(UPDATED_AT);
+        }
+      };
```

Comment:
Nit:
Could be simplified to 
```suggestion
  private static final Set<String> OUTER_COLUMNS = Set.of(CREATED_AT, ID, UPDATED_AT);
```